app/src/main/java/com/lastaoutdoor/lasta/ui/screen/social/ConversationScreen.kt

```diff
@@ -152,19 +180,111 @@ fun ShowMessage(message: MessageModel, user: UserModel) {
     textColor = MaterialTheme.colorScheme.onPrimary
   }
 
+  // Check if the message is a shared activity by checking if it starts with "|$@!|"
+  val isActivityShared = message.content.startsWith("|$@!|")
+  val messageParts = message.content.split("|")
+  val activityName = if (isActivityShared) messageParts[3] else message.content
+  val activityId = if (isActivityShared) messageParts[2] else ""
+  val activityType =
+      if (isActivityShared) ActivityType.valueOf(messageParts[4]) else ActivityType.HIKING
+  val activityDifficulty =
+      if (isActivityShared) Difficulty.valueOf(messageParts[5]) else Difficulty.EASY
+
   Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = arrangement) {
-    Card(
-        modifier =
-            Modifier.padding(8.dp)
-                .widthIn(0.dp, LocalConfiguration.current.screenWidthDp.dp * 0.55f),
-        colors =
-            CardColors(
-                containerColor = backgroundColor,
-                contentColor = textColor,
-                disabledContentColor = backgroundColor,
-                disabledContainerColor = textColor,
-            )) {
-          Text(message.content, modifier = Modifier.padding(8.dp))
-        }
+    if (isActivityShared) {
+      // Display the activity card
+      ActivityCard(
+          activityId,
+          activityName,
+          activityType,
+          activityDifficulty,
+          changeActivityToDisplay,
+          navigateToMoreInfo) // Display the ActivityCard
+    } else {
+      Card(
+          modifier =
+              Modifier.padding(8.dp)
+                  .widthIn(0.dp, LocalConfiguration.current.screenWidthDp.dp * 0.55f),
+          colors =
+              CardColors(
+                  containerColor = backgroundColor,
+                  contentColor = textColor,
+                  disabledContentColor = backgroundColor,
+                  disabledContainerColor = textColor,
+              )) {
+            Text(text = message.content, modifier = Modifier.padding(8.dp))
+          }
+    }
   }
 }
+
+/** Display the activity card of a shared activity */
+@Composable
+fun ActivityCard(

```

Comment:
I feel like this activity card could still be further improved for a more appealing/natural interface. 

![image](https://github.com/LASTA-OUTDOOR/lasta/assets/95919886/1b8e8236-cc4f-449e-8754-373604d92730)

From the screenshot, I believe that it could be smaller in height, also the gray background could be replaced by a light blue one. Also, the arrow could be rather at the right corner of the rectangled, maybe surrounded  by a circle so it looks like a true button

Reply:
This is a nice suggestion I can picture it, will definitely implement next sprint if time is on our side !